src/haiway/helpers/files.py

```
@@ -0,0 +1,388 @@
+    create: bool,
+    exclusive: bool,
+) -> int:
+    if not path.exists():
+        if create:
+            path.parent.mkdir(parents=True, exist_ok=True)
+            path.touch()
+
+        else:
+            raise FileException(f"File does not exist: {path}")
+
+    file_handle: int = open(path, "r+b").fileno()
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)